test/testPartyAttachmentsAfterUpdate.js

```diff
@@ -0,0 +1,659 @@
+    'filename',
+    'originalFilename',
+    'tmpFileName',
+    'size',
+    'hash',
+  ]
+  expectedFileProperties.forEach(prop => {
+    if (file.file[prop] === undefined) {
+      throw new SriError({
+        status: 500,
+        errors: [
+          {
+            code: "file.file.is.missing.expected.property",
+            type: "ERROR",
+            message: `expected file.file object is missing property ${prop}`,
+          },
+        ],
+      });
+    }
+  })
+
+  if (file.attachment === null || typeof file.attachment !== 'object') {
+    throw new SriError({
+      status: 500,
+      errors: [
+        {
+          code: "file.file.is.not.an.object",
+          type: "ERROR",
+          message: `expected file.attachment to be an object`,
+        },
+      ],
+    });
+  }
+  const expectedAttachmentProperties = [
+    'key',
+    'description',
+    'aCustomTestProperty',
+  ]
+  expectedAttachmentProperties.forEach(prop => {
+    if (file.attachment[prop] === undefined) {
+      throw new SriError({
+        status: 500,
+        errors: [
+          {
+            code: "file.attachment.is.missing.expected.property",
+            type: "ERROR",
+            message: `expected file.attachment object is missing property ${prop}`,
+          },
+        ],
+      });
+    }
+  })
+
+  if (file.resource === null || typeof file.resource !== 'object') {
+    throw new SriError({
+      status: 500,
+      errors: [
+        {
+          code: "file.resource.is.not.an.object",
+          type: "ERROR",
+          message: `expected resource.attachment to be an object`,
+        },
+      ],
+    });
+  }
+  const expectedResourceProperties = [
+    'href',
+    'key',
+  ]
+  expectedResourceProperties.forEach(prop => {
+    if (file.resource[prop] === undefined) {
+      throw new SriError({
+        status: 500,
+        errors: [
+          {
+            code: "file.resource.is.missing.expected.property",
+            type: "ERROR",
+            message: `expected file.attachment object is missing property ${prop}`,
+          },
+        ],
+      });
+    }
+  })
+}
+
+
+/**
+ * 
+ * @param {*} handleMultipleUploadsTogether 
+ * @param {*} uploadInSequence 
+ * @param {*} checkStoreAttachmentsReceivedList 
+ * @returns 
+ */
+function checkStoreAttachmentFactory(handleMultipleUploadsTogether, uploadInSequence, checkStoreAttachmentsReceivedList = undefined) {
+  if (handleMultipleUploadsTogether) {
+    return (files) => files.forEach(file => checkStoreAttachment(file))
+  } else {
+    if (uploadInSequence) {
+      return (file) => {
+        checkStoreAttachmentsReceivedList.push(file.fileObj.filename.substring(0,8));
+        checkStoreAttachment(file);
+      }
+    } else {
+      return checkStoreAttachment
+    }
+  }
+}
+
+function clearCheckStoreAttachmentsReceivedList(checkStoreAttachmentsReceivedList) {
+  if (checkStoreAttachmentsReceivedList) {
+    checkStoreAttachmentsReceivedList.length = 0; // clear the array
```

Comment:
`checkStoreAttachmentsReceivedList = []` would remove the need for an exist-check.